### PR TITLE
Mk 1724 add multisig tests

### LINKED ISSUE #1724: add more Manager_lambda scenarios to contract multisig integration tests
The contract multisig integration tests can be expanded to include also tests for
  setDelegate,   removeDelegate,   transferToContract,  in addition to the current   transferImplicit,

adding the three tests.

### PR DESCRIPTION
**Pull request by michaelkernaghan**
CLOSES #1724 

added tests:

contract-multisig-remove-delegate.spec.ts
contract-multisig-set-delegate.spec.ts
contract-multisig-transfer-to-contract.spec.ts

### CODE REVIEW

File: integration-tests/contract-multisig-set-delegate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        "0",
+        // Sub function
+        'operation',
+        // Action
+        MANAGER_LAMBDA.setDelegate(pkh),
+        // Signature list
+        [signature1.prefixSig, signature2.prefixSig, null]
+      ).send()
+      await op3.confirmation();
+    })
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-set-delegate.spec.ts:134]: Should the test validate that the delegate is indeed set, and that the operation was successful? 
Is it the smart contract that will be set pkh as a delegate? The test can validate the the delegate of the contract is indeed pkh.

File: integration-tests/contract-multisig-transfer-implicit.spec.ts [hunk 2]
```diff
@@ -4,14 +4,13 @@ import { genericMultisig } from "./data/multisig";
...
   const Tezos = lib;
-  const test = require('jest-retries')
 
   describe(`Generic Multisig: ${rpc}`, () => {
     beforeEach(async (done) => {
       await setup()
       done()
     })
-    test('test manager transfers scenarios', 2, async (done: () => void) => {
+    test('test manager transfers scenarios', async (done: () => void) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-transfer-implicit.spec.ts:13]: This part can be removed as well: `: () => void`

File: integration-tests/contract-multisig-transfer-implicit.spec.ts [hunk 3]
```diff
@@ -123,6 +122,9 @@ CONFIGS().forEach(({ lib, rpc, setup, createAddress }) => {
       ).send()
 
       await op2.confirmation();
+      console.log("op2")
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-transfer-implicit.spec.ts:125]: Console.log can be removed

File: integration-tests/contract-multisig-transfer-to-contract.spec.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        'operation',
+        // Action
+        MANAGER_LAMBDA.transferToContract(knownContract, 1),
+        // Signature list
+        [signature1.prefixSig, signature2.prefixSig, null]
+      ).send()
+
+      await op2.confirmation();
+      expect(op2.status).toEqual('applied')
+
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-transfer-to-contract.spec.ts:126]: Consider validating that the balance of knownContract increased by 1

File: integration-tests/data/extended_multisig.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+export const extended_multisig = `{ parameter
```
Comment (roxaneletourneau) [inline integration-tests/data/extended_multisig.ts:1]: The file doesn't seem to be used and could be deleted

File: integration-tests/contract-multisig-transfer-implicit.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply integration-tests/contract-multisig-transfer-implicit.spec.ts:13]: removed

File: integration-tests/contract-multisig-transfer-to-contract.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply integration-tests/contract-multisig-transfer-to-contract.spec.ts:126]: I have added validation for the before and after contract balances.

File: integration-tests/data/extended_multisig.ts (hunk 5 above)
Comment (michaelkernaghan) [reply integration-tests/data/extended_multisig.ts:1]: whoops, from another project. Deleted :)

File: integration-tests/contract-multisig-remove-delegate.spec.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
+import { CONFIGS } from "./config";
+import { MANAGER_LAMBDA } from "@taquito/taquito";
+import { genericMultisig } from "./data/multisig";
+
+CONFIGS().forEach(({ lib, rpc, setup, createAddress }) => {
+  const Tezos = lib;
+ 
+  describe(`Generic Multisig set delegate: ${rpc}`, () => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-multisig-remove-delegate.spec.ts:8]: oops copy/paste mistake.  fixed.

File: integration-tests/contract-multisig-remove-delegate.spec.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        'operation',
+        // Action
+        MANAGER_LAMBDA.removeDelegate(),
+        // Signature list
+        [signature1.prefixSig, signature2.prefixSig, null]
+      ).send()
+
+      await op3.confirmation();
+
+      expect(delegate).toEqual(null)
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-remove-delegate.spec.ts:130]: Should a delegate be set first, and then removed? If calling removeDelegate on a contract with no delegate, we might not be testing the correct behavior. 

A suggestion could be to combine the `integration-tests/contract-multisig-set-delegate.spec.ts` and `integration-tests/contract-multisig-remove-delegate.spec.ts` into one test where we 
1. set a delegate 
2. confirm that the delegate is updated
3. remove the delegate
4. confirmed that the delegate is removed

File: integration-tests/contract-multisig-set-delegate.spec.ts [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      done()
+    })
+    test('test manager transfers set delegate scenarios', async () => {
+
+      const account1 = await createAddress();
+      const account2 = await createAddress();
+      const account3 = await createAddress();
+
+      const pkh = await Tezos.signer.publicKeyHash();
+      const op = await Tezos.contract.registerDelegate({});
```
Comment (roxaneletourneau) [inline integration-tests/contract-multisig-set-delegate.spec.ts:20]: Wondering if lines 20 to 23 are necessary for the current test

File: integration-tests/contract-multisig-remove-delegate.spec.ts (hunk 7 above)
Comment (michaelkernaghan) [reply integration-tests/contract-multisig-remove-delegate.spec.ts]: i have rewritten the test to work this way and removed the separate removal test.

File: integration-tests/contract-multisig-set-delegate.spec.ts (hunk 8 above)
Comment (michaelkernaghan) [reply integration-tests/contract-multisig-set-delegate.spec.ts]: i have to say the pkh is a delegate beofre i can get the contract to add it as delegate through lambda

Comment (roxaneletourneau) [reply integration-tests/contract-multisig-set-delegate.spec.ts]: oh yes, make sense!

**Review by roxaneletourneau — APPROVED** [review]
Looks good to me!

**Pull request merged**